lib/IRGen/IRGenSIL.cpp

```diff
@@ -737,6 +737,8 @@ class IRGenSILFunction :
       return;
 
     llvm::IRBuilder<> ZeroInitBuilder(AI->getNextNode());
+
+    // No debug location is how LLVM marks prologue instructions.
```

Comment:
I'm still hazy on how function prologues are determined, and why we have to be careful about them w.r.t debug locations at the IR level. Feel free to chime in on https://reviews.llvm.org/D47097: I listed my questions in more detail over there, and it would help out our GSoC intern.